facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java

```diff
@@ -0,0 +1,170 @@
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnvironmentProviderTest  extends AbstractProviderTest<BuildEnvironment>{
+
+    @Mock
+    private BuildEnvironmentRepository repository;
+
+    @Spy
+    @InjectMocks
+    private EnvironmentProviderImpl provider;
+
+    @Override
+    protected AbstractProvider provider() {
+        return provider;
+    }
+
+    @Override
+    protected Repository<BuildEnvironment, Integer> repository() {
+        return repository;
+    }
+
+    BuildEnvironment env = prepareBuildEnvironment("Crash Test Dummy 1");
+
+    @Before
+    public void fill() {
+        List<BuildEnvironment> envs = new ArrayList<>();
+        envs.add(prepareBuildEnvironment("Hello"));
+        envs.add(prepareBuildEnvironment("My name is"));
+        envs.add(prepareBuildEnvironment("S-S-S-Slim Shady"));
+        envs.add(env);
+        fillRepository(envs);
+    }
+
+    @Test
+    public void testGetAll() {
+        Page<Environment> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(4)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(env.getName()), "Environment present"));
+    }
+
+    @Test
+    public void testStore() {
+        final String name = "Hello";
+        final String sysImageId = "NewID";
+
+        Environment environment = Environment.builder().name(name).systemImageId(sysImageId).systemImageType(SystemImageType.DOCKER_IMAGE).build();
+
+        Environment stored = provider.store(environment);
+
+        assertThat(stored.getName()).isEqualTo(name);
+        assertThat(stored.getId()).isNotNull();
+        assertThat(stored.getSystemImageId()).isEqualTo(sysImageId);
+        assertThat(stored.getSystemImageType()).isEqualTo(SystemImageType.DOCKER_IMAGE);
+        Mockito.verify(repository).save(ArgumentMatchers
+                .argThat(env -> name.equals(env.getName())));
+    }
+
+    @Test
+    public void testGetSpecific() {
+        Environment environment = provider.getSpecific(env.getId().toString());
+
+        assertThat(environment.getId()).isEqualTo(env.getId().toString());
+        assertThat(environment.getName()).isEqualTo(env.getName());
+    }
+
+    @Test
+    public void testStoreWithId(){
+        final String name = "Hello";
+        final String sysImageId = "sysID";
+
+        Environment environment = Environment.builder().id("3113").name(name).systemImageId(sysImageId).systemImageType(SystemImageType.DOCKER_IMAGE).build();
+
+        try{
+            provider.store(environment);
+            fail("Validation exception expected");
+        }catch(DTOValidationException ex){
+            // ok
+        }
+    }
+
+    @Test
+    public void testUpdate(){
```

Comment:
I think we don't allow updating of environments via REST.

Reply:
YEP